zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -102,7 +111,16 @@ class EcommerceSpider(Args[EcommerceSpiderParams], BaseSpider):
     @classmethod
     def from_crawler(cls, crawler: Crawler, *args, **kwargs) -> scrapy.Spider:
         spider = super(EcommerceSpider, cls).from_crawler(crawler, *args, **kwargs)
-        spider.allowed_domains = [get_domain(spider.args.url)]
+        if spider.args.use_url_lists:
+            response = requests.get(spider.args.url)
```

Comment:
Rather than using an external library, why don't we use Scrapy downloader handlers? 
That would also open the door for fetching s3 / FTP urls. I implemented something like that in the past:
https://github.com/VMRuiz/GenericCrawlerPOC/blob/main/generic_crawler/input.py

Reply:
On my example, calling the handlers skips the middlewares, so offsite requests should not be an issue. 
Although I admit that the resulting code is a little bit complex and collides with Zyte API usage if enabled.

Reply:
Just to clarify, the issue with allowed domains is that we set it automatically based on start URLs from `from_crawler`. If we need to post-pone the fetching of start URLs to the asynchronous `start_requests`, by the time that method is called the offsite middleware will have already processed the list of allowed domains.

We could hack around it (find the middleware instance and edit it from `start_requests`), it is just rather ugly. Or we can *not* set a limit on allowed domains to begin with when loading lists of URLs.